python/pyplugin_installer/installer.py

```diff
@@ -555,6 +555,10 @@ def installFromZipFile(self, filePath):
 
         pluginFileName = os.path.splitext(os.path.basename(filePath))[0]
 
+        if not pluginName:
+            QMessageBox.warning(iface.mainWindow(), self.tr("QGIS Python Install from ZIP Plugin Installer"), self.tr('The plugin directory was not found inside the ZIP file. You should create a plugin folder, put the files inside and create the ZIP file again.'), QMessageBox.Ok)
```

Comment:
@DelazJ can you check that for the wording?

I would suggest to use something like "The ZIP file is not a valid QGIS python plugin. No root folder was found inside." (i.e.  I would not assume that the one who installs the plugin is the one who developed the plugin).

Reply:
Ping @DelazJ

Reply:
Oups, Sorry! I forgot that one. Though not sure how my wording in English can be better than yours, @m-kuhn :thinking: .
Anyways...  I agree that the person who installs is not necessarily the developer, and because of that I think we should give them some hints (what's a valid plugin structure?) to avoid them unnecessarily wait for the next point release of the plugin. And my personal use of this tool was to download a branch from github so I guess, no fix to expect from the dev and as far as i remember I had to manually rename things to get it work. Maybe we need some magic in the tool to have a proper name or, keep your rewording with a "more info" that would point to https://docs.qgis.org/3.4/en/docs/user_manual/plugins/plugins.html#the-install-from-zip-tab where we could explain what a proper structure of zip file should be.
Not sure whether i'm not out of the scope of this PR... :thinking: :

Reply:
@DelazJ I think this would be great (add more info), if apart from the message is explained to the user what is the desired structure for the zip, could be achieved in a next attempt not make the same mistake again.

I can change the wording like @m-kuhn says and i can make the docs to explain the desired structure, we can do this in this PR if you think that is ok.